File: examples/vbnet/07_ComplexStructures.vb
Imports StackExchange.Redis
Imports System

Module Program
    Sub Main(args As String())
        Dim redis = ConnectionMultiplexer.Connect("localhost")
        Dim db = redis.GetDatabase()

        Console.WriteLine("--- 07 Complex Structures ---")

        ' List
        db.ListRightPush("cart", "apple")
        db.ListRightPush("cart", "banana")
        Dim item = db.ListLeftPop("cart")
        Console.WriteLine($"Removed from cart: {item}")

        ' Set
        db.SetAdd("colors", "red")
        db.SetAdd("colors", "blue")
        Dim isRed As Boolean = db.SetContains("colors", "red")
        Console.WriteLine($"Contains red? {isRed}")

        ' Hash
        db.HashSet("config:ui", New HashEntry() {
            New HashEntry("theme", "dark"),
            New HashEntry("fontsize", "14px")
        })
        Dim theme = db.HashGet("config:ui", "theme")
        Console.WriteLine($"Theme: {theme}")
    End Sub
End Module

File: examples/csharp/07_ComplexStructures.cs
using StackExchange.Redis;
using System;

class Program
{
    static void Main(string[] args)
    {
        var redis = ConnectionMultiplexer.Connect("localhost");
        var db = redis.GetDatabase();

        Console.WriteLine("--- 07 Complex Structures ---");

        // List
        db.ListRightPush("logs", "Error: File not found");
        db.ListRightPush("logs", "Info: User logged in");
        Console.WriteLine($"Logs count: {db.ListLength("logs")}");

        // Set
        db.SetAdd("admins", "user:1");
        db.SetAdd("admins", "user:2");
        bool isAdmin = db.SetContains("admins", "user:1");
        Console.WriteLine($"Is user:1 admin? {isAdmin}");

        // Hash
        db.HashSet("user:1", new HashEntry[] { 
            new HashEntry("name", "Alice"), 
            new HashEntry("age", 30) 
        });
        string name = db.HashGet("user:1", "name");
        Console.WriteLine($"User Name: {name}");
    }
}
